src/steps/MatchColumnsStep/MatchColumnsStep.tsx

```diff
@@ -68,7 +68,8 @@ export const MatchColumnsStep = <T extends string>({ data, headerValues, onConti
   const { fields, autoMapHeaders, autoMapDistance, translations } = useRsi<T>()
   const [isLoading, setIsLoading] = useState(false)
   const [columns, setColumns] = useState<Columns<T>>(
-    (headerValues as string[]).map((value, index) => ({ type: ColumnType.empty, index, header: value })),
+    // Do not remove spread, it indexes empty array elements, otherwise map() skips over them
+    ([...headerValues] as string[]).map((value, index) => ({ type: ColumnType.empty, index, header: value ?? "" })),
```

Comment:
Could you double check if spread is necessary

Reply:
yep, you can try it out in your console 
`[,,1,2].map(e =>({a:1})) -> [empty × 2, {…}, {…}]v`
vs
`[...[,,2,2]].map(e =>({a:e})) -> [{…}, {…}, {…}, {…}]`